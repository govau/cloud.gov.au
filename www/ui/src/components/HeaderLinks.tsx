import * as React from "react";
import styled from "../styled-components";
import { space, responsiveStyle } from "styled-system";

import BaseLink, { LinkProps, RouterLinkProps } from "./Link";

const mediaDisplay = responsiveStyle({
  cssProperty: "display"
});

const Ul = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`;

interface LiProps {
  display: string[];
  m: number[];
}

const UnstyledLi: React.StatelessComponent<
  LiProps & React.HTMLProps<HTMLLIElement>
> = ({ children, ...rest }) => <li {...rest}>{children}</li>;

const StyledLi = styled(UnstyledLi)`
  ${mediaDisplay};
  ${space};
`;

const Li: React.StatelessComponent<{}> = ({ children }) => (
  <StyledLi display={["block", "inline-block"]} m={[1]}>
    {children}
  </StyledLi>
);

const UnstyledLink: React.StatelessComponent<
  Partial<LinkProps & RouterLinkProps>
> = ({ children, ...rest }) => <BaseLink {...rest}>{children}</BaseLink>;

const Link = styled(UnstyledLink)``;

interface Props {}

const HeaderLinks: React.StatelessComponent<Props> = () => (
  <Ul>
    <Li>
      <Link href="https://docs.cloud.gov.au/">Docs</Link>
    </Li>
  </Ul>
);

export default HeaderLinks;
