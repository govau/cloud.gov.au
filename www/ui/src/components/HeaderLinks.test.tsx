import * as React from "react";
import { MemoryRouter } from "react-router";
import * as renderer from "react-test-renderer";

import HeaderLinks from "./HeaderLinks";

it("renders without crashing", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <HeaderLinks />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
