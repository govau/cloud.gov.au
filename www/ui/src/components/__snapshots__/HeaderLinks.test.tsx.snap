// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c3 {
  -webkit-transition: 0.25s;
  transition: 0.25s;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 1px;
}

.c2 {
  -webkit-transition: 0.25s;
  transition: 0.25s;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 1px;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c1 {
  display: block;
  margin: 8px;
}

@media screen and (min-width:40em) {
  .c1 {
    display: inline-block;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
    display={
      Array [
        "block",
        "inline-block",
      ]
    }
    m={
      Array [
        1,
      ]
    }
  >
    <a
      className="c2"
      href="/code/"
      onClick={[Function]}
    >
      Code
    </a>
  </li>
  <li
    className="c1"
    display={
      Array [
        "block",
        "inline-block",
      ]
    }
    m={
      Array [
        1,
      ]
    }
  >
    <a
      className="c3"
      href="https://docs.cloud.gov.au/"
    >
      Docs
    </a>
  </li>
  <li
    className="c1"
    display={
      Array [
        "block",
        "inline-block",
      ]
    }
    m={
      Array [
        1,
      ]
    }
  >
    <a
      className="c3"
      href="https://support.cloud.gov.au/"
    >
      Support
    </a>
  </li>
</ul>
`;
